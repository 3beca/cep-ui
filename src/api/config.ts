export const BASE_URL = process.env.REACT_APP_BASE_URL || 'https://cep.tribeca.ovh';
export const EVENT_TYPE_URL = '/admin/event-types';