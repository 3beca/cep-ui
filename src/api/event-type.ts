import {
    fetchData,
    RequestInfo,
    ResponseData,
    ErrorAPI,
    ResponseEmptyData
} from './fetch-data';
import {
    BASE_URL,
    EVENT_TYPE_URL
} from './config';
export * from './fetch-data';

export type EventType = {
    name: string;
    id: string;
    url: string;
    createdAt: string;
    updatedAt: string;
};
export type EventTypeList = {
    results: EventType[];
    next?: string | undefined;
    prev?: string | undefined;
};
export const getEventTypeList = (config: RequestInfo) => async (page: number = 1, size: number = 10): Promise<ResponseData<EventTypeList>|ErrorAPI> => {
    const url = `${config.url}${EVENT_TYPE_URL}/?page=${page}&pageSize=${size}`;
    return await fetchData<undefined, EventTypeList>({...config, url, method: 'GET'});
};
export const deleteEventType = (config: RequestInfo) => async (eventTypeIds: string|string[]): Promise<(ResponseEmptyData|ErrorAPI)[]> => {
    const eventTypesArray = (Array.isArray(eventTypeIds)) ? eventTypeIds : [eventTypeIds];
    if(!eventTypeIds || eventTypesArray.length < 1) {
        return [{
            status: 500,
            error: 'Missing EventType id',
            message: 'eventTypeIds is an invalid EventTypeId value or array'
        }];
    }
    const requests = [];
    for (const eventTypeId of eventTypesArray) {
        const url = `${config.url}${EVENT_TYPE_URL}/${eventTypeId}`;
        requests.push(fetchData<undefined, undefined>({...config, url, method: 'DELETE'}));
    }
    return Promise.all(requests);
};
export type EventTypeApi = {
    getEventTypeList(page?: number, size?: number): Promise<ResponseData<EventTypeList>|ErrorAPI>;
    deleteEventType(eventTypeIds: string|string[]): Promise<(ResponseEmptyData|ErrorAPI)[]>;
};
export const EventTypeApi = (): EventTypeApi => {
    const config: RequestInfo = {
        url: BASE_URL,
        method: 'GET',
        headers: {}
    };
    return {
        getEventTypeList: getEventTypeList(config),
        deleteEventType: deleteEventType(config)
    };
};
