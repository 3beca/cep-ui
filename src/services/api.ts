import {
    fetchApi,
    APIRequestInfo,
    APIResponseData,
    APIError,
    isAPIError
} from '../utils/fetch-api';

export type ServiceList<T> = {
    results: T[];
    next?: string | undefined;
    prev?: string | undefined;
};
export type ServiceError = {
    statusCode: number,
    error: string,
    message: string
};
export type Entity = {
    id: string;
};
export type WithDate = {
    createdAt: string;
    updatedAt: string;
};
export type ServiceDeleted = {
    state: 'DELETED'|'REJECTED';
    error?: ServiceError;
} & Entity;
export type EventType = {
    name: string;
    url: string;
} & Entity & WithDate;
export type EventTypeList = ServiceList<EventType>;
export type EventTypeError = ServiceError;
export type EventTypeDeleted = ServiceDeleted;

export type Target = {
    name: string;
    url: string;
} & Entity & WithDate;
export type TargetList = ServiceList<Target>;
export type TargetError = ServiceError;
export type TargetDeleted = ServiceDeleted;

export type RuleTypes = 'sliding'|'hopping'|'tumbling'|'none';
export type Rule = {
    name: string;
    type: RuleTypes;
    targetId: string;
    eventTypeId: string;
    skipOnConsecutivesMatches?: boolean;
    filters: any;
} & Entity & WithDate;
export type RuleList = ServiceList<Rule>;
export type RuleError = ServiceError;
export type RuleDeleted = ServiceDeleted;

export type EventLog = {
    eventTypeId: string;
    eventTypeName: string;
    payload: {[key: string]: any;};
    requestId: string;
    createdAt: string;
    rules: {
        id: string;
        name: string;
        targetId: string;
    }[];
    targets: {
        id: string;
        name: string;
        response: {
            statusCode: number;
            body: {[key: string]: any;};
        };
    }[];
} & Entity;

export const getListRequest = (baseURL: string, config: APIRequestInfo) => async <T>(path: string, page: number = 1, size: number = 10, filter: string = ''): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}/?page=${page}&pageSize=${size}${filter ? `&search=${filter}` : ''}`;
    return fetchApi<undefined, ServiceList<T>, ServiceError>(url, {...config, method: 'GET'});
};
export const deleteRequest = (baseURL: string, config: APIRequestInfo) => async (path: string, eventTypeIds: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)> => {
    const eventTypesArray = ((Array.isArray(eventTypeIds)) ? eventTypeIds : [eventTypeIds]).filter(id => !!id);

    if(!eventTypeIds || eventTypesArray.length < 1) {
        return {
            errorCode: 500,
            errorMessage: '',
            error: {
                statusCode: 500,
                error: 'Missing id',
                message: 'id is an invalid id value or array'
            }
        };
    }
    const requests = [];
    for (const eventTypeId of eventTypesArray) {
        const url = `${baseURL}${path}/${eventTypeId}`;
        requests.push(fetchApi<undefined, undefined, ServiceError>(url, {...config, method: 'DELETE'}));
    }
    const responses = await Promise.all(requests);
    const data: ServiceDeleted[] = responses.map((response, idx) => {
        const deletedResponse: ServiceDeleted = {
            id: eventTypesArray[idx],
            state: 'DELETED'
        };
        if(isAPIError(response)) {
            deletedResponse.state = 'REJECTED';
            deletedResponse.error = response.error;
        }
        return deletedResponse;
    });
    return {
        status: 200,
        data
    };
};
export type Api = {
    getListRequest<T>(path: string, page?: number, size?: number, filter?: string): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>>;
    deleteRequest(path: string, ids: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)>;
};
export const buildApiService = (server: string): Api => {
    const config: APIRequestInfo = {
        method: 'GET',
        headers: {}
    };
    return {
        getListRequest: getListRequest(server, config),
        deleteRequest: deleteRequest(server, config)
    };
};
