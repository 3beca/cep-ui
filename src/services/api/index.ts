import {
    fetchApi,
    APIRequestInfo,
    APIResponseData,
    APIError,
    isAPIError,
    APIBody
} from '../../utils/fetch-api';
import {
    ServiceList,
    ServiceError,
    ServiceDeleted,
} from './models';
export * from './models';

export const getListRequest = (baseURL: string, config: APIRequestInfo) => async <T>(path: string, page: number = 1, size: number = 10, filter: string = ''): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}/?page=${page}&pageSize=${size}${filter ? `&search=${filter}` : ''}`;
    return fetchApi<undefined, ServiceList<T>, ServiceError>(url, {...config, method: 'GET'});
};
export const deleteRequest = (baseURL: string, config: APIRequestInfo) => async (path: string, entityIds: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)> => {
    const entityArray = ((Array.isArray(entityIds)) ? entityIds : [entityIds]).filter(id => !!id);

    if(!entityIds || entityArray.length < 1) {
        return {
            errorCode: 500,
            errorMessage: '',
            error: {
                statusCode: 500,
                error: 'Missing id',
                message: 'id is an invalid id value or array'
            }
        };
    }
    const requests = [];
    for (const entityId of entityArray) {
        const url = `${baseURL}${path}/${entityId}`;
        requests.push(fetchApi<undefined, undefined, ServiceError>(url, {...config, method: 'DELETE'}));
    }
    const responses = await Promise.all(requests);
    const data: ServiceDeleted[] = responses.map((response, idx) => {
        const deletedResponse: ServiceDeleted = {
            id: entityArray[idx],
            state: 'DELETED'
        };
        if(isAPIError(response)) {
            deletedResponse.state = 'REJECTED';
            deletedResponse.error = response.error;
        }
        return deletedResponse;
    });
    return {
        status: 200,
        data
    };
};
export const createRequest = (baseURL: string, config: APIRequestInfo) => async <T extends APIBody>(path: string, body: Partial<T>): Promise<APIResponseData<T>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}`;
    return fetchApi<Partial<T>, T, ServiceError>(url, {...config, method: 'POST', headers: {...config.headers, 'content-type': 'application/json'}, body});
};
export type Api = {
    getListRequest<T>(path: string, page?: number, size?: number, filter?: string): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>>;
    deleteRequest(path: string, ids: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)>;
    createRequest<T extends APIBody>(path: string, body: Partial<T>): Promise<APIResponseData<T>|APIError<ServiceError>>;
};
export const buildApiService = (server: string): Api => {
    const config: APIRequestInfo = {
        method: 'GET',
        headers: {}
    };
    return {
        getListRequest: getListRequest(server, config),
        deleteRequest: deleteRequest(server, config),
        createRequest: createRequest(server, config)
    };
};
