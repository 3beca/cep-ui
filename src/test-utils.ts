import nock from 'nock';

export const setupNock = (url: string) => {
    const server = nock(url).defaultReplyHeaders({ 'access-control-allow-origin': '*' });
    // Skip Preflight CORS OPTION request
    nock(url).intercept(/./, 'OPTIONS').reply(200, undefined, { 'access-control-allow-origin': '*' }).persist();
    return server;
}