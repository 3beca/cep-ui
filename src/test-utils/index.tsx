import * as React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory, History} from 'history';
import {render} from '@testing-library/react';
import {MainMenuProvider} from '../services/main-menu-provider';
export * from '@testing-library/react';
export * from './api';

type OptionsRenderRouter = {
    route?: string;
    history?: History
};
export function renderInsideApp(
    ui: React.ReactElement,
    {
        route = '/',
        history = createMemoryHistory({initialEntries: [route]}),
        ...renderOptions
    }: OptionsRenderRouter = {}
) {
    const Wrapper: React.FC<{}> = function Wrapper(props) {
        return (
            <Router history={history}>
                <MainMenuProvider {...props} />
            </Router>
        );
    }
    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions}),
        history
    };
};
