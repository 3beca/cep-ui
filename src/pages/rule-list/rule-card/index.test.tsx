import * as React from 'react';
import userEvent from '@testing-library/user-event';
import {Link} from 'react-router-dom';
import { RuleTypes } from '../../../services/api';
import RuleCard, {colorTypeSelector, mapRuleTypeName} from './index';
import { generateRule, render, screen } from '../../../test-utils';

const fakeLink = Link as unknown as {linkAction: jest.Mock};
jest.mock('react-router-dom', () => {
    const React = require('react');
    const linkAction = jest.fn();
    const Link = React.forwardRef((props: any, ref: any) => {
        return <div ref={ref} {...props} onClick={() => {
            props.onClick && props.onClick();
            linkAction(props.to);
        }}/>
    });
    Link.linkAction = linkAction;
    return {
        Link
    };
});

beforeEach(() => fakeLink.linkAction.mockClear());

test('selector avatar color function shuold return a default color', () => {
    expect(colorTypeSelector(undefined as unknown as RuleTypes, {ruleCardAvatarPurple: 'default color'} as any)).toEqual('default color');
});

test('mapRuleTypeName should return REAL TIME when undefined type', () => {
    expect(mapRuleTypeName(undefined as unknown as RuleTypes)).toEqual('REAL TIME');
});

test('mapRuleTypeName should return REAL TIME when invalid type', () => {
    expect(mapRuleTypeName('invalidtype' as unknown as RuleTypes)).toEqual('REAL TIME');
});

test('RuleCard should render with a rule and snap', () => {
    const rule = generateRule('rule-test', 3, {'_or': [{'hola': 1}], '_and': [{'adios': 2}], p1: 'valor'});
    const {container} = render(<RuleCard rule={rule}/>);

    expect(screen.getByLabelText(/avatar icon/)).toHaveTextContent('R');
    expect(container).toMatchSnapshot();
});

test('RuleCard should render with a eventType name, target name, filters and skip execution disable', async () => {
    const rule = generateRule('rule-test', 3, {'_or': [{'hola': 1}], '_and': [{'adios': 2}], p1: 'valor'});
    const {container} = render(<RuleCard rule={rule}/>);

    expect(await screen.findByLabelText(/avatar icon/)).toHaveTextContent('R');
    expect(await screen.findByLabelText(/eventType name card rule/)).toHaveTextContent('EventType 3-rule-test');
    expect(await screen.findByLabelText(/target name card rule/)).toHaveTextContent('Target 3-rule-test');
    await screen.findByLabelText(/filters card rule/);
    expect(await screen.findByLabelText(/status card rule/)).toHaveTextContent(/skip consecutives/i);
    await screen.findByLabelText(/skip consecutives disable/);
    expect(await screen.findByRole('checkbox')).toHaveAttribute('readonly');
    expect(container).toMatchSnapshot();
});

test('RuleCard should render with a eventType name, target name, filter passthrow and skip execution enable', async () => {
    const rule = generateRule('rule-test', 2, {});
    const {container} = render(<RuleCard rule={rule}/>);

    expect(await screen.findByLabelText(/avatar icon/)).toHaveTextContent('T');
    expect(await screen.findByLabelText(/eventType name card rule/)).toHaveTextContent('EventType 2-rule-test');
    expect(await screen.findByLabelText(/target name card rule/)).toHaveTextContent('Target 2-rule-test');
    expect(await screen.findByLabelText(/filters card rule/)).toHaveTextContent(/^filters.*passthrow$/i);
    expect(await screen.findByLabelText(/status card rule/)).toHaveTextContent(/skip consecutives/i);
    await screen.findByLabelText(/skip consecutives enable/);
    expect(await screen.findByRole('checkbox')).toHaveAttribute('readonly');
    expect(container).toMatchSnapshot();
});

test('RuleCard should open the context menu and navigate to details', async () => {
    const rule = generateRule('rule-test', 2, {});
    render(<RuleCard rule={rule}/>);
    await screen.findByLabelText(/setting dialog card rule hidden$/);

    expect(await screen.findByLabelText(/avatar icon/)).toHaveTextContent('T');
    userEvent.click(await screen.findByLabelText(/settings card rule$/i));
    await screen.findByLabelText(/setting dialog card rule visible/);

    userEvent.click(await screen.findByLabelText(/setting dialog details card rule$/i));
    expect(fakeLink.linkAction).toHaveBeenCalledTimes(1);
    expect(fakeLink.linkAction).toHaveBeenNthCalledWith(1, '/rules/details/2_rule-test');
    await screen.findByLabelText(/setting dialog card rule hidden$/);
});