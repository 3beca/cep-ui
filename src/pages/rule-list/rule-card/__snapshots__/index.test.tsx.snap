// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleCard should render with a eventType name, target name, filter passthrow and skip execution enable 1`] = `
<div>
  <div
    aria-label="element card rule"
    class="MuiPaper-root MuiCard-root makeStyles-ruleCard-275 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-label="header rule card"
      class="MuiCardHeader-root makeStyles-ruleCardHeader-276"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="avatar icon"
          class="MuiAvatar-root MuiAvatar-circle makeStyles-ruleCardAvatar-288 makeStyles-ruleCardAvatarOrange-292 MuiAvatar-colorDefault"
        >
          T
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Rule 2
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          1/1/2020 11:10:10
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      aria-label="content card rule"
      class="makeStyles-ruleCardContent-277"
    >
      <div
        aria-label="eventType name card rule"
        class="makeStyles-ruleCardBody-278"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-279 MuiTypography-body1"
        >
          Event Type:
        </p>
        <p
          aria-label="eventType name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-280 MuiTypography-body1"
        >
          EventType 2-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="target name card rule"
        class="makeStyles-ruleCardBody-278"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-279 MuiTypography-body1"
        >
          Target:
        </p>
        <p
          aria-label="target name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-280 MuiTypography-body1"
        >
          Target 2-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="filters card rule"
        class="makeStyles-ruleCardFilters-281"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-279 MuiTypography-body1"
        >
          Filters:
        </p>
        <div
          aria-label="filters container"
        >
          <div
            aria-label="filter expression passthrow"
            class="makeStyles-ruleCardFilterExpressionPassthrow-396"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              PASSTHROW
            </h6>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="status card rule"
        class="makeStyles-ruleCardStatus-282"
      >
        <div
          class="makeStyles-ruleCardStatusDate-283"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
        </div>
        <div
          class="makeStyles-ruleCardStatusOneShot-284"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Skip Consecutives
          </p>
          <div
            aria-label="skip consecutives enable"
            class="makeStyles-onShotLabel-285 makeStyles-onShotLabelOn-286"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RuleCard should render with a eventType name, target name, filters and skip execution disable 1`] = `
<div>
  <div
    aria-label="element card rule"
    class="MuiPaper-root MuiCard-root makeStyles-ruleCard-138 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-label="header rule card"
      class="MuiCardHeader-root makeStyles-ruleCardHeader-139"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="avatar icon"
          class="MuiAvatar-root MuiAvatar-circle makeStyles-ruleCardAvatar-151 makeStyles-ruleCardAvatarPurple-154 MuiAvatar-colorDefault"
        >
          R
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Rule 3
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          1/1/2020 11:10:10
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      aria-label="content card rule"
      class="makeStyles-ruleCardContent-140"
    >
      <div
        aria-label="eventType name card rule"
        class="makeStyles-ruleCardBody-141"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-142 MuiTypography-body1"
        >
          Event Type:
        </p>
        <p
          aria-label="eventType name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-143 MuiTypography-body1"
        >
          EventType 3-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="target name card rule"
        class="makeStyles-ruleCardBody-141"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-142 MuiTypography-body1"
        >
          Target:
        </p>
        <p
          aria-label="target name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-143 MuiTypography-body1"
        >
          Target 3-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="filters card rule"
        class="makeStyles-ruleCardFilters-144"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-142 MuiTypography-body1"
        >
          Filters:
        </p>
        <div
          aria-label="filters container"
        >
          <div
            aria-label="container expressions"
            class="makeStyles-ruleCardFilterContainer-256"
          >
            <div
              aria-label="container expressions header"
              class="makeStyles-ruleCardFilterContainerHeader-257"
            >
              <p
                class="MuiTypography-root makeStyles-ruleCardFilterContainerHeaderText-258 MuiTypography-body1"
              >
                One Of
              </p>
            </div>
            <div
              aria-label="filter expression comparator"
              class="makeStyles-ruleCardFilterExpression-263"
            >
              <p
                aria-label="filter expression field"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-260 MuiTypography-body1"
              >
                hola
              </p>
              <p
                aria-label="filter expression operator"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-261 MuiTypography-body1"
              >
                =
              </p>
              <p
                aria-label="filter expression value"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-262 MuiTypography-body1"
              >
                1
              </p>
            </div>
          </div>
          <div
            aria-label="container expressions"
            class="makeStyles-ruleCardFilterContainer-256"
          >
            <div
              aria-label="container expressions header"
              class="makeStyles-ruleCardFilterContainerHeader-257"
            >
              <p
                class="MuiTypography-root makeStyles-ruleCardFilterContainerHeaderText-258 MuiTypography-body1"
              >
                All Of
              </p>
            </div>
            <div
              aria-label="filter expression comparator"
              class="makeStyles-ruleCardFilterExpression-263"
            >
              <p
                aria-label="filter expression field"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-260 MuiTypography-body1"
              >
                adios
              </p>
              <p
                aria-label="filter expression operator"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-261 MuiTypography-body1"
              >
                =
              </p>
              <p
                aria-label="filter expression value"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-262 MuiTypography-body1"
              >
                2
              </p>
            </div>
          </div>
          <div
            aria-label="filter expression comparator"
            class="makeStyles-ruleCardFilterExpression-263"
          >
            <p
              aria-label="filter expression field"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-260 MuiTypography-body1"
            >
              p1
            </p>
            <p
              aria-label="filter expression operator"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-261 MuiTypography-body1"
            >
              =
            </p>
            <p
              aria-label="filter expression value"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-262 MuiTypography-body1"
            >
              valor
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="status card rule"
        class="makeStyles-ruleCardStatus-145"
      >
        <div
          class="makeStyles-ruleCardStatusDate-146"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
        </div>
        <div
          class="makeStyles-ruleCardStatusOneShot-147"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Skip Consecutives
          </p>
          <div
            aria-label="skip consecutives disable"
            class="makeStyles-onShotLabel-148 makeStyles-onShotLabelOff-150"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RuleCard should render with a rule and snap 1`] = `
<div>
  <div
    aria-label="element card rule"
    class="MuiPaper-root MuiCard-root makeStyles-ruleCard-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-label="header rule card"
      class="MuiCardHeader-root makeStyles-ruleCardHeader-2"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <div
          aria-label="avatar icon"
          class="MuiAvatar-root MuiAvatar-circle makeStyles-ruleCardAvatar-14 makeStyles-ruleCardAvatarPurple-17 MuiAvatar-colorDefault"
        >
          R
        </div>
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Rule 3
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          1/1/2020 11:10:10
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      aria-label="content card rule"
      class="makeStyles-ruleCardContent-3"
    >
      <div
        aria-label="eventType name card rule"
        class="makeStyles-ruleCardBody-4"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-5 MuiTypography-body1"
        >
          Event Type:
        </p>
        <p
          aria-label="eventType name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-6 MuiTypography-body1"
        >
          EventType 3-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="target name card rule"
        class="makeStyles-ruleCardBody-4"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-5 MuiTypography-body1"
        >
          Target:
        </p>
        <p
          aria-label="target name"
          class="MuiTypography-root makeStyles-ruleCardBodyName-6 MuiTypography-body1"
        >
          Target 3-rule-test
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="filters card rule"
        class="makeStyles-ruleCardFilters-7"
      >
        <p
          class="MuiTypography-root makeStyles-ruleCardBodyLabel-5 MuiTypography-body1"
        >
          Filters:
        </p>
        <div
          aria-label="filters container"
        >
          <div
            aria-label="container expressions"
            class="makeStyles-ruleCardFilterContainer-119"
          >
            <div
              aria-label="container expressions header"
              class="makeStyles-ruleCardFilterContainerHeader-120"
            >
              <p
                class="MuiTypography-root makeStyles-ruleCardFilterContainerHeaderText-121 MuiTypography-body1"
              >
                One Of
              </p>
            </div>
            <div
              aria-label="filter expression comparator"
              class="makeStyles-ruleCardFilterExpression-126"
            >
              <p
                aria-label="filter expression field"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-123 MuiTypography-body1"
              >
                hola
              </p>
              <p
                aria-label="filter expression operator"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-124 MuiTypography-body1"
              >
                =
              </p>
              <p
                aria-label="filter expression value"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-125 MuiTypography-body1"
              >
                1
              </p>
            </div>
          </div>
          <div
            aria-label="container expressions"
            class="makeStyles-ruleCardFilterContainer-119"
          >
            <div
              aria-label="container expressions header"
              class="makeStyles-ruleCardFilterContainerHeader-120"
            >
              <p
                class="MuiTypography-root makeStyles-ruleCardFilterContainerHeaderText-121 MuiTypography-body1"
              >
                All Of
              </p>
            </div>
            <div
              aria-label="filter expression comparator"
              class="makeStyles-ruleCardFilterExpression-126"
            >
              <p
                aria-label="filter expression field"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-123 MuiTypography-body1"
              >
                adios
              </p>
              <p
                aria-label="filter expression operator"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-124 MuiTypography-body1"
              >
                =
              </p>
              <p
                aria-label="filter expression value"
                class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-125 MuiTypography-body1"
              >
                2
              </p>
            </div>
          </div>
          <div
            aria-label="filter expression comparator"
            class="makeStyles-ruleCardFilterExpression-126"
          >
            <p
              aria-label="filter expression field"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionField-123 MuiTypography-body1"
            >
              p1
            </p>
            <p
              aria-label="filter expression operator"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionOperator-124 MuiTypography-body1"
            >
              =
            </p>
            <p
              aria-label="filter expression value"
              class="MuiTypography-root makeStyles-ruleCardFilterExpressionValue-125 MuiTypography-body1"
            >
              valor
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        aria-label="status card rule"
        class="makeStyles-ruleCardStatus-8"
      >
        <div
          class="makeStyles-ruleCardStatusDate-9"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
        </div>
        <div
          class="makeStyles-ruleCardStatusOneShot-10"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Skip Consecutives
          </p>
          <div
            aria-label="skip consecutives disable"
            class="makeStyles-onShotLabel-11 makeStyles-onShotLabelOff-13"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
